Cython/Utility/ModuleSetupCode.c

```diff
@@ -556,6 +556,19 @@
   #define __PYX_IS_UNSIGNED(type) (((type)-1) > 0)
 #endif
 
+#if CYTHON_COMPILING_IN_PYPY == 1
+  #if PY_VERSION_HEX >= 0x030800b4 && PY_VERSION_HEX < 0x030A0000
+    #define CYTHON_NEED_TP_PRINT 1
+  #else
+    #define CYTHON_NEED_TP_PRINT 0
+  #endif
+#else
+  #if PY_VERSION_HEX >= 0x030800b4 && PY_VERSION_HEX < 0x03090000
+    #define CYTHON_NEED_TP_PRINT 1
+  #else
+    #define CYTHON_NEED_TP_PRINT 0
+  #endif
+#endif
```

Comment:
Either, we consider this a feature-flag, in which case it should go into the CPYTHON/PYPY/… target define blocks further up (and be definable/overridable by users), or we consider this something internal, then it shouldn't start with `CYTHON_` but with `__PYX_`.

I'm not sure if it's worth making this visible to users. The slot is unlikely to reappear in the future and for historical Python versions, we should be able to control it ourselves correctly. Thus, `__PYX_` seems better to me.

You can also collapse the conditional defines to
```c
#define __PYX_NEED_TP_PRINT_SLOT  (PY_VERSION_HEX >= 0x030800b4 && PY_VERSION_HEX < 0x03090000)
```

Reply:
done, thanks